executor/batch_checker.go

```diff
@@ -146,8 +146,30 @@ func getKeysNeedCheckOneRow(ctx sessionctx.Context, t table.Table, row []types.D
 		}
 	}
 
-	// addChangingColTimes is used to fetch values while processing "modify/change column" operation.
-	addChangingColTimes := 0
+	// extraColumns is used to fetch values while processing "add/drop/modify/change column" operation.
+	extraColumns := 0
+	for _, col := range t.WritableCols() {
+		// if there is a changing column, append the dependency column for index fetch values
+		if col.ChangeStateInfo != nil && col.State != model.StatePublic {
+			// I'm not sure why this value doesn't need to be casted
```

Comment:
I think we need to cast the value here. Please help add it？

Reply:
Good. Done.